compositor_common/src/scene.rs

```diff
@@ -0,0 +1,37 @@
+use std::{any::Any, collections::HashMap, rc::Rc};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct VideoId(usize);
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct TransformationRegistryKey(String);
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct Resolution {
+    width: usize,
+    height: usize,
+}
+
+#[derive(Debug)]
+pub enum Node {
+    Video {
+        id: VideoId,
+        resolution: Resolution,
+    },
+
+    Image {
+        data: Vec<u8>,
+        resolution: Resolution,
+    },
+
+    Transformation {
+        registry_key: TransformationRegistryKey,
+        inputs: HashMap<String, Rc<Node>>,
```

Comment:
This object will be constructed in response to some API call, but values will be read inside the renderer (on a different thread), so unless you are planning to construct and keep this object only in render thread then everything here needs to an Arc

Reply:
true, I looked at the `Arc`s I had here yesterday, thought they're not needed and replaced them with `Rc`s... 😅